openedx/core/djangoapps/import_export/courses/views.py

```diff
@@ -0,0 +1,529 @@
+                    name
+                )
+            else:
+                export_course_to_xml(
+                    modulestore(),
+                    contentstore(),
+                    courselike_module.id,
+                    root_dir,
+                    name
+                )
+
+            logging.debug(
+                u'tar file being generated at {0}'.format(export_file.name)
+            )
+            with tarfile.open(name=export_file.name, mode='w:gz') as tar_file:
+                tar_file.add(root_dir / name, arcname=name)
+        except SerializationError as exc:
+            log.exception(
+                u'There was an error exporting course %s',
+                courselike_key
+            )
+            unit = None
+            failed_item = None
+            parent = None
+            try:
+                failed_item = modulestore().get_item(exc.location)
+                parent_loc = modulestore().get_parent_location(
+                    failed_item.location
+                )
+
+                if parent_loc is not None:
+                    parent = modulestore().get_item(parent_loc)
+                    if parent.location.category == 'vertical':
+                        unit = parent
+            except:  # pylint: disable=bare-except
+                # if we have a nested exception, then we'll show the more
+                # generic error message
+                pass
+
+            return self._export_error_response(
+                {
+                    "context_course": str(courselike_module.location),
+                    "error": True,
+                    "error_message": str(exc),
+                    "failed_module":
+                    str(failed_item.location) if failed_item else None,
+                    "unit":
+                    str(unit.location) if unit else None
+                },
+                redirect_url=redirect_url
+            )
+        except Exception as exc:
+            log.exception(
+                'There was an error exporting course %s',
+                courselike_key
+            )
+            return self._export_error_response(
+                {
+                    "context_course": courselike_module.url_name,
+                    "error": True,
+                    "error_message": str(exc),
+                    "unit": None
+                },
+                redirect_url=redirect_url
+            )
+        finally:
+            shutil.rmtree(root_dir / name)
+
+        # The course is all set; return the tar.gz
+        wrapper = FileWrapper(export_file)
+
+        response = HttpResponse(wrapper, content_type='application/x-tgz')
+        response['Content-Disposition'] = 'attachment; filename={}'.format(
+            os.path.basename(
+                export_file.name.encode('utf-8')
+            )
+        )
+        response['Content-Length'] = os.path.getsize(export_file.name)
+        return response
+
+    # pylint: disable=unused-argument
```

Comment:
You can prefix variable names with an underscore to indicate you know they are unused. Please do that rather than disable this warning.